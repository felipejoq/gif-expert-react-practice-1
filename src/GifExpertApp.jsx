import {useState} from "react";
import {AddCategory, GifGrid} from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball']);

    const onAddCategory = (newCategory) => {

        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories])
    }

    return (
        <>
            <h1>Gif Expert App</h1>

            <AddCategory
                onNewCategory={valueNewCategory => onAddCategory(valueNewCategory)}
            />

            {
                categories.map((category) => (
                    <GifGrid
                        category={category}
                        key={category}
                    />
                ))
            }
        </>
    )
}