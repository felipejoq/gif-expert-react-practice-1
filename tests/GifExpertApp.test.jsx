import {fireEvent, render, screen} from "@testing-library/react";
import {GifExpertApp} from "../src/GifExpertApp.jsx";

describe('Pruebas en <GifExpertApp />', () => {

    const inputValue = 'Hey Arnold';

    test('Debe hacer match con el snapshot', () => {
        const {container} = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    })

    test('Debe crear un nuevo resultado al enviar el formulario.', () => {
        render(<GifExpertApp />);
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target: { value: inputValue } });
        fireEvent.submit(form);

        expect(screen.getByText(inputValue)).toBeTruthy();
        expect(input.value).toBe('');
        // screen.debug();
    });


});