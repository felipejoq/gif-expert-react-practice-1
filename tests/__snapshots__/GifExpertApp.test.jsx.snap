// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    Dragon Ball
  </h3>
  <h4>
    Cargando...
  </h4>
  <div
    class="card-grid"
  />
</div>
`;
